detox/src/configuration/composeSessionConfig.js

```diff
@@ -1,16 +1,20 @@
+const _ = require('lodash');
+
 const isValidWebsocketURL = require('../utils/isValidWebsocketURL');
+const log = require('../utils/logger').child({ cat: 'config' });
 
 /**
  * @param {{
  *  cliConfig: Record<string, any>;
  *  globalConfig: Detox.DetoxConfig;
  *  localConfig: Detox.DetoxConfiguration;
  *  errorComposer: import('../errors/DetoxConfigErrorComposer');
+ *  isCloudSession: Boolean
  * }} options
  */
 async function composeSessionConfig(options) {
-  const { errorComposer, cliConfig, globalConfig, localConfig } = options;
-
+  const { errorComposer, cliConfig, globalConfig, localConfig, isCloudSession } = options;
```

Comment:
[MUST HAVE]
Authentication is absolutely mandatory on cloud.

QQ: Is there an easy way to detect device type here ? If yes, then we don't need this extra param `isCloudSession`.